internal/commands/secrets/update_inputs_test.go

```diff
@@ -0,0 +1,83 @@
+package secrets
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestResolveSecret(t *testing.T) {
+	testLen := 7
+	secrets := make([]realm.Secret, testLen)
+
+	for i := 0; i < testLen; i++ {
+		secrets[i] = realm.Secret{
+			fmt.Sprintf("secretID%d", i),
+			fmt.Sprintf("secretName%d", i),
+		}
+	}
+
+	for _, tc := range []struct {
+		description    string
+		testInput      string
+		expectedOutput realm.Secret
+	}{
+		{
+			description:    "should return the secret if searching by name",
+			testInput:      "secretName4",
+			expectedOutput: secrets[4],
+		},
+		{
+			description:    "should return the secret if searching by ID",
+			testInput:      "secretID3",
+			expectedOutput: secrets[3],
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			inputs := updateInputs{
+				secret: tc.testInput,
+			}
+			result, err := inputs.resolveSecret(nil, secrets)
+			assert.Nil(t, err)
+			assert.Equal(t, tc.expectedOutput, result)
+		})
+	}
+
+	t.Run("should return an error if we cannot find the secret to update", func(t *testing.T) {
+		inputs := updateInputs{
+			secret: "cannotFind",
+		}
+
+		result, err := inputs.resolveSecret(nil, secrets)
+		assert.Equal(t, realm.Secret{}, result)
+		assert.Equal(t, "unable to find the secret with the identifier cannotFind", err.Error())
+	})
+
+	t.Run("should show the prompt for a user if the input is empty", func(t *testing.T) {
+		inputs := updateInputs{}
+
+		_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+		assert.Nil(t, consoleErr)
+		defer console.Close()
+
+		doneCh := make(chan struct{})
+		go func() {
+			defer close(doneCh)
+			console.ExpectString("Which secret would you like to update?")
+			console.SendLine("secretID2")
+			console.SendLine(" ")
```

Comment:
I'd be surprised both `SendLine` calls are needed here, can you test around with this?  i think with `Select`, you just need to hit "enter" to confirm your selection (as opposed to the `MultiSelect`), so this 2nd line here may not be doing anything